model/task_history.go

```diff
@@ -351,16 +352,12 @@ func (iter *taskHistoryIterator) GetChunk(v *Version, numBefore, numAfter int, i
 	return chunk, nil
 }
 
-func (self *taskHistoryIterator) GetDistinctTestNames(numCommits int) ([]string, error) {
-	session, mdb, err := db.GetGlobalSessionFactory().GetSession()
-	if err != nil {
-		return nil, errors.Wrap(err, "problem getting database session")
-	}
-	defer session.Close()
-
-	session.SetSocketTimeout(time.Minute)
-
-	pipeline := mdb.C(task.Collection).Pipe(
+func (self *taskHistoryIterator) GetDistinctTestNames(env evergreen.Environment, numCommits int) ([]string, error) {
+	ctx, cancel := env.Context()
```

Comment:
Is it possible to instead thread through the request's context? This way the query would get cancelled when the request is killed.
With the environment's context I think the query will continue even after the request is cancelled, until we hit max time.

Reply:
Just a note that it doesn't actually kill the query on the server (but it may still be a good idea to do so). https://mongodb.slack.com/archives/C7WJZNUTA/p1612824042065600

Reply:
Oh wow. In that case, is there a point to making this change?

Reply:
It looks like context doesn't do anything, but SetMaxTime does (but won't work on context being cancelled). Do we know how long the context being cancelled takes? In which case we could set SetMaxTime for slightly under that. I'm also considering implementing May's solution but I don't fully understand it yet.

Reply:
I think the context is controlled by how long the API consumer decides to wait. A reasonable max would probably be slightly above the timeout for the load balancer, since from the caller's perspective a higher max won't be any different

Reply:
> May's solution

May's solution seems like it's well suited for being integrated into anser.

Reply:
I decided to just embed it, since I'm not yet sure if it's going to be an improvement and we may want to iterate on this.